src/com/lilithsthrone/game/character/GameCharacter.java

```diff
@@ -2887,8 +3007,421 @@ public boolean isSlave() {
+	 */
+	public void removeRule(PlayerSlaveryRule ruleToRemove)
+	{
+		if(ruleToRemove != null)
+		{
+			this.slaveOwnerRules.remove(ruleToRemove);
+		}
+	}
+	
+	/**
+	 * Checks to see if a slavery rule is enforced by the owner. If it is enforced, returns it; otherwise returns null.
+	 * @param ruleUniqueName
+	 * @return
+	 */
+	public PlayerSlaveryRule hasRule(String ruleUniqueName)
+	{
+		for(PlayerSlaveryRule slaveryRule : this.slaveOwnerRules) {
+			if(slaveryRule.getUniqueName() == ruleUniqueName)
+			{
+				return slaveryRule;
+			}
+		}
+		return null;
+	}
+	
+	/**
+	 * Checks to see if a slavery rule is enforced by the owner. If it is enforced, returns it; otherwise returns null.
+	 * @return
+	 */
+	public PlayerSlaveryRule hasRule(PlayerSlaveryRule ruleToFind)
+	{
+		for(PlayerSlaveryRule slaveryRule : this.slaveOwnerRules) {
+			if(slaveryRule.getUniqueName() == ruleToFind.getUniqueName())
+			{
+				return slaveryRule;
+			}
+		}
+		return null;
+	}
+	
+	/**
+	 * Returns the full list of rules the owner has.
+	 */
+	public Set<PlayerSlaveryRule> getRules()
+	{
+		return this.slaveOwnerRules;
+	}
+	
+	/**
+	 * Returns true if the character is within the bounds of their owner's property.
+	 */
+	public boolean isWithinOwnersPropery()
+	{
+		if(this.getOwner() == null)
+		{
+			return true;
+		}
+		for(Value<WorldType, Vector2i> locationValue : this.getOwner().getSlaveAreaLocationPlaces())
+		{
+			if(this.getWorldLocation() == locationValue.getKey() && this.getLocation().getX() == locationValue.getValue().getX() && this.getLocation().getY() == locationValue.getValue().getY())
+			{
+				return true;
+			}
+		}
+		return false;
+	}
+	
+	/**
+	 * Automatically enters the scene that triggers when player passes out from their collar or gets brought back by enforcers.
+	 * 
+	 * It should handle the movement to a new location as well.
+	 */
+	public void triggerSlaveRetrivalDialogue()
+	{
+		//Main.game.getPlayer().setLocation(this.worldLocation, this.location, false);
+		//DialogueNodeOld dn = null; //todo
+		
+		// Warning: We are doing dangerous shit there. Since slavery system is supposed to be an overbearing one, it's safe to do it here, but to other contributors, here's a note - don't do it without a good reason!
+		// To explain - this function below sets the dialogue node.
+		// It can force it in the middle of anything, potentially sequence breaking a quest dialogue.
+		// Using it outside of these tight constraints should be avoided.
+		// (Note for Inno: I'm just leaving this as a warning for future contributors :3)
+		//Main.game.setContent(new Response("", "", dn));		
+	}
+	
+	/**
+	 * Returns true if character is wearing a clothing piece enchanted with enslaving or contraband enslaving enchantmnet.
+	 * @return
+	 */
+	public boolean isWearingEnslavingClothing()
+	{
+		return this.getClothingCurrentlyEquipped().stream().anyMatch(e -> e.isContrabandEnslavementClothing() || e.isEnslavementClothing());
```

Comment:
ooh streams (ignore me, I'm just gonna go scroll up and point out some streams for you)